### PR TITLE
[camera_avfoundation] fix sample times not being numeric after pause/resume.

### PR DESCRIPTION
**Pull request by misos1**
Correctly initialise `_lastVideoSampleTime` and `_lastAudioSampleTime` before first use to avoid propagating invalid (non-numeric) values into methods like `appendPixelBuffer`.

Fixes https://github.com/flutter/flutter/issues/132014

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/camera/camera_avfoundation/ios/Classes/FLTCam.m
Comment (hellohuanlin) [inline packages/camera/camera_avfoundation/ios/Classes/FLTCam.m:678]: are these already set when we pause?

Comment (misos1) [reply packages/camera/camera_avfoundation/ios/Classes/FLTCam.m:678]: The problem was happening because if pause and resume happened before the first sample buffer arrived then they would not be properly initialised when first used, they would have zero timescale/denominator and `CMTimeAdd` and `CMTimeSubtract` propagate that "non-numeric" value through `_videoTimeOffset` into `nextSampleTime` (or if this is not the first recording then they will contain some old timestamp from previous recording).

Comment (hellohuanlin) [reply packages/camera/camera_avfoundation/ios/Classes/FLTCam.m:678]: Interesting. It isn't clear from the code. Can you add this reasoning as a comment in code?

**Review by hellohuanlin — APPROVED** [review]

**Comment by stuartmorgan** [comment]
@hellohuanlin Please request a second review when reviewing a PR that needs two approvals.

**Review by bparrishMines — APPROVED** [review]
LGTM

**Pull request merged**
